eZ/Publish/Core/FieldType/User/Type.php

```diff
@@ -243,7 +263,50 @@ public function validate(FieldDefinition $fieldDefinition, SPIValue $fieldValue)
             return $errors;
         }
 
-        if (!$fieldValue->hasStoredLogin) {
+        if (!is_string($fieldValue->login) || empty($fieldValue->login)) {
+            $errors[] = new ValidationError(
+                'Login required',
+                null,
+                [],
+                'username'
+            );
+        }
+
+        if (!is_string($fieldValue->email) || empty($fieldValue->email)) {
+            $errors[] = new ValidationError(
+                'Email required',
+                null,
+                [],
+                'email'
+            );
+        } elseif (!preg_match(self::EMAIL_FORMAT_PATTERN, $fieldValue->email)) {
```

Comment:
I'm not the biggest fan of custom validation for e-mail, urls and so on. There are built in PHP functions for some basic checks: https://www.php.net/manual/en/filter.examples.validation.php#filter.examples.validation